src/purpleLayer.js

```diff
@@ -63,11 +63,14 @@ L.LayerGroup.PurpleLayer = L.LayerGroup.extend(
         getMarker: function (data) {
               var lat = data.Lat;
               var lng = data.Lon;
-              var value = parseFloat(data.PM2_5Value);  //PM2.5 VALUE in microgram per metre cube
 
+              var value = parseFloat(data.PM2_5Value);  //PM2.5 VALUE in microgram per metre cube
+              if(! (lat || lng || value) ){
```

Comment:
Can you put an extra space before the `{`?
I believe it would be clearer if you move the content before the `if` to a var, something like:

```js
var isLocationPresent = lat || lng || value;

if (!isLocationPresent) {
  return;
}
```

What do you think?